### PR TITLE
Remove references to validation set in evaluator

### LINKED ISSUE #1474: Refactor the concept of public and private test set
Auto-sklearn was built for the AutoML challenges where there were public and private test sets by the names "validation set" and "test set". Since these challenges we have substantially refactored Auto-sklearn and no longer allow the user to pass in the public test set aka "validation set". This leads to confusion as "validation set" is really a misnomer, as inside a machine learning system it usually denotes the dataset on which a model is selected. Therefore we should drop all references that still exist to this old concept of a "validation set", as already being done by @eddiebergman in #1434, which should only be in the `evaluation` submodule`.

### PR DESCRIPTION
**Pull request by eddiebergman**
* Closes #1474 
---
* Just search up `X_valid`, `y_valid`, `validation` and cleaned up where I thought it's needed
* The test file for evaluators is difficult to parse and uses a lot of mocks. There was a lot of magic numbers I just changed to what the tests ended up giving back

### CODE REVIEW

File: test/test_evaluation/test_abstract_evaluator.py [hunk 1]
```diff
@@ -92,46 +90,6 @@ def test_finish_up_model_predicts_NaN(self):
...
-            file_output=True,
-            status=StatusType.SUCCESS,
-        )
-        self.assertEqual(loss, 1.0)
-        self.assertEqual(
-            additional_run_info,
-            {"error": "Model predictions for validation set " "contains NaNs."},
-        )
-
-        # NaNs in prediction test
```
Comment (mfeurer) [inline test/test_evaluation/test_abstract_evaluator.py:115]: We could keep the testing of the `test` set, right?

Comment (eddiebergman) [reply test/test_evaluation/test_abstract_evaluator.py:115]: The one above this comment is for validation but sure, the one below I could just remove the `predictions_valid` part and put it back in

**Comment by eddiebergman** [comment]
@mfeurer Once these tests pass, it's ready for merge

**Review by mfeurer — APPROVED** [review]
